### PR TITLE
Split web and management servers

### LINKED ISSUE #15908: Split REST server from management server
**Description**

The goal is to split the management server and Spring Boot web server, such that the web server will be usable for the REST API. 

As part of this issue, we need to:

- [x] Specify a port 8080 for the server, i.e. `server.port = 8080`
- [x] Specify a port for the management server, i.e. `management.server.port = 9600`
- [x] Ensure the management endpoints are _only_ available via the management port
- [x] Ensure the testing infrastructure (e.g. `TestStandaloneBroker`, `TestStandaloneGateway`, etc) are using random/unique ports for the new server port
- [x] The Dockerfile is extended to add the new port
- [x] Ensure the old broker health redirects (`/ready`, `/health`, `/startup`) are only applied on the management server port

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

This PR introduces a new Spring web server on port 8080 (the default port used by Spring, and also the default for all the webapps). This effectively causes the management server to be split to port 9600, which introduces a new Spring web context (you can use the combo of `ConditionalOnManagementContext` and `ManagementContextConfiguration` annotations to do things specifically there).

This means a few things:

- Actuator endpoints _must_ be defined using the annotations found in the actuate package, e.g. `RestControllerEndpoint`, `WebEndpoint`, Endpoint`, etc. These are _only_ picked up by the management server.
- Normal REST endpoints should use the web stereotypes, e.g. `RestController`, `Controller`, etc. These will _only_ be picked up by the main web server.

If you want to add `WebFilter`, `RouterFunction`, etc., for a specific context, make sure to use the `ConditionalOnManagementContext` and `ManagementContextConfiguration` combo (and add your configuration class to the imports file).

Additionally, the PR ensures our existing test infrastructure works out of the box by ensuring the new port is randomized every time, allowing multiple concurrent instances to run simultaneously.

## Related issues

closes #15908 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: dist/src/main/java/io/camunda/zeebe/shared/MainSupport.java
Comment (npepinpe) [inline dist/src/main/java/io/camunda/zeebe/shared/MainSupport.java:40]: :thought_balloon: We were explicitly reducing the number of threads because the management server didn't need so many. But in this case, I don't know how many we need for the REST API, so let's just leave it to the defaults.

File: dist/src/test/java/io/camunda/zeebe/ForbidWebStereotypeTest.java
Comment (npepinpe) [inline dist/src/test/java/io/camunda/zeebe/ForbidWebStereotypeTest.java:36]: :thought_balloon: This is more to enforce that REST stuff should live in the REST module, and actuators (which should live in dist) should not use those annotations since otherwise it will be loaded in the REST server.

File: dist/src/test/java/io/camunda/zeebe/ForbidWebStereotypeTest.java
Comment (koevskinikola) [inline dist/src/test/java/io/camunda/zeebe/ForbidWebStereotypeTest.java:20]: :x: Did you forget to finish this Javadoc, or am I missing something?

**Review by koevskinikola — APPROVED** [review]
LGTM :tada: 

Just a small feedback for a Javadoc that seems unfinished and may be confusing to understand. But it doesn't impact the correctness of the PR code, so I'm approving it.

File: dist/src/test/java/io/camunda/zeebe/ForbidWebStereotypeTest.java
Comment (npepinpe) [reply dist/src/test/java/io/camunda/zeebe/ForbidWebStereotypeTest.java:20]: Yeah, just unfinished :sweat_smile:

**Comment by npepinpe** [comment]
This PR relies on #15913, so cannot merge until then.

**Pull request merged**
